crd/apis/network/v1/gkenetworkparamset_types.go

```diff
@@ -64,14 +64,77 @@ type NetworkRanges struct {
+// particular GKENetworkParamSet condition type has been raised.
+type GKENetworkParamSetReason string
+
+const (
+	// SubnetNotFound indicates that the specified subnet was not found.
+	SubnetNotFound GKENetworkParamSetReason = "SubnetNotFound"
+	// SecondaryRangeNotFound indicates that the specified secondary range was not found.
+	SecondaryRangeNotFound GKENetworkParamSetReason = "SecondaryRangeNotFound"
+	// DeviceModeCantBeUsedWithSecondaryRange indicates that device mode was used with a secondary range.
+	DeviceModeCantBeUsedWithSecondaryRange GKENetworkParamSetReason = "DeviceModeCantBeUsedWithSecondaryRange"
+	// DeviceModeVPCAlreadyInUse indicates that the VPC is already in use by another GKENetworkParamSet resource.
+	DeviceModeVPCAlreadyInUse GKENetworkParamSetReason = "DeviceModeVPCAlreadyInUse"
+	// DeviceModeCantUseDefaultVPC indicates that a device mode GKENetworkParamSet cannot use the default VPC.
+	DeviceModeCantUseDefaultVPC GKENetworkParamSetReason = "DeviceModeCantUseDefaultVPC"
+	// DPDKUnsupported indicates that DPDK device mode is not supported on the current cluster.
+	DPDKUnsupported GKENetworkParamSetReason = "DPDKUnsupported"
+)
+
+// GNPNetworkParamsReadyReason defines the set of reasons that explains
+// the ParamsReady condition on the referencing Network resource.
+type GNPNetworkParamsReadyReason string
+
+const (
+	// L3SecondaryMissing indicates that the L3 type Network resource is
+	// referencing a GKENetworkParamSet with secondary range unspecified.
+	L3SecondaryMissing GNPNetworkParamsReadyReason = "L3SecondaryMissing"
+	// L3DeviceModeExists indicates that the L3 type Network resource is
+	// referencing a GKENetworkParamSet with device mode specified.
+	L3DeviceModeExists GNPNetworkParamsReadyReason = "L3DeviceModeExists"
+	// DeviceModeMissing indicates that the Device type Network resource is
+	// referencing a GKENetworkParamSet with device mode unspecified.
+	DeviceModeMissing GNPNetworkParamsReadyReason = "DeviceModeMissing"
+	// DeviceSecondaryExists indicates that the Device type Network resource is
+	// referencing a GKENetworkParamSet with a secondary range specified.
+	DeviceSecondaryExists GNPNetworkParamsReadyReason = "DeviceSecondaryExists"
+)
+
 // GKENetworkParamSetStatus contains the status information related to the network.
 type GKENetworkParamSetStatus struct {
 	// PodCIDRs specifies the CIDRs from which IPs will be used for Pod interfaces
 	// +optional
 	PodCIDRs *NetworkRanges `json:"podCIDRs,omitempty"`
 
 	// Conditions is a field representing the current conditions of the GKENetworkParamSet.
-	Conditions []metav1.Condition `json:"conditions"`
+	//
+	// Known condition types are:
+	//
+	// * "Ready"
+	//
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
```

Comment:
is `protobuf:"bytes,1,rep,name=conditions"` valid here ? I am not familiar with this notation

Reply:
its from your example https://github.com/mskrocki/kubernetes/commit/89c14ae71a7f841d8eadc93a38879bf77eebf962#diff-f3bb8746d632dccbc31c8a7761219fb639bcf18a16f646a6fc5d08dfbbb56cb6R894